### PR TITLE
[FLINK-33042][flamegraph] Allow trigger flamegraph when task is initializing

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What is the purpose of the change

Currently, the flamegraph can be triggered when task is running.

After [FLINK-17012](https://issues.apache.org/jira/browse/FLINK-17012) and [FLINK-22215](https://issues.apache.org/jira/browse/FLINK-22215), flink split the running to running and initializing. We should allow trigger flamegraph when task is initializing. For example, the initialization is very slow, we need to troubleshoot.

Here is a stack example, task is rebuilding the rocksdb after the parallelism is changed.

image

## Brief change log

[FLINK-33042][flamegraph] Allow trigger flamegraph when task is initializing

## Verifying this change

Added the `@Parameter public ExecutionState executionState;` for the `VertexThreadInfoTrackerTest`. And all tests will run with running and initializing.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Review by RocMarshal — APPROVED** [review]
Hi, @1996fanrui Thanks for the contribution and ping.
I checked the `state switching conditions` of `Task` and test cases diff. 
LGTM +1. 👍

**Review by 1996fanrui — COMMENTED** [review]
Thanks @RocMarshal for your review!

After I run it locally with INITIALIZING subtasks, and I found some code of front end need to be updated as well. I have updated them, please check again in your free time, thanks!

File: flink-runtime-web/web-dashboard/src/app/pages/job/overview/flamegraph/job-overview-drawer-flamegraph.component.html [hunk 1]
```diff
@@ -16,8 +16,13 @@
...
 
-<span *ngIf="selectedVertex?.detail?.status !== 'RUNNING'">
-  Operator is not running. Cannot sample back pressure.
+<span
+  *ngIf="
+    selectedVertex?.detail?.status !== 'RUNNING' &&
+    selectedVertex?.detail?.status !== 'INITIALIZING'
+  "
+>
+  Operator is not running or initializing. Cannot sample flame graph.
```
Comment (RocMarshal) [inline flink-runtime-web/web-dashboard/src/app/pages/job/overview/flamegraph/job-overview-drawer-flamegraph.component.html:25]: 👍  nice~

**Review by RocMarshal — APPROVED** [review]
Hi, @1996fanrui ,I checked on my PC with `long-time restore/initializing` and `running` .
LGTM +1.
Glad to see the better usability on flame graph.
Thank you very much~

**Review by 1996fanrui — COMMENTED** [review]
Thank you @RocMarshal  for your hard test, merging~

**Pull request merged**